src/Toml/Parser/Value.hs

```diff
@@ -22,13 +22,31 @@ import Control.Applicative.Combinators (between, count, option, optional, sepBy1
 import Data.Fixed (Pico)
 import Data.Time (Day, LocalTime (..), TimeOfDay, ZonedTime (..), fromGregorianValid,
                   makeTimeOfDayValid, minutesToTimeZone)
+import Data.Char (ord)
+
 import Text.Read (readMaybe)
 
-import Toml.Parser.Core (Parser, binary, char, digitChar, hexadecimal, lexeme, octal, sc, signed,
+import Toml.Parser.Core (Parser, char, digitChar, hexDigitChar, octDigitChar, binDigitChar, lexeme, sc, signed,
                          string, text, try, (<?>))
 import Toml.Parser.String (textP)
 import Toml.Type (AnyValue, UValue (..), typeCheck)
 
+-- | Returns decimal value from base N representation
+baseStringToInt :: Integer -> String -> Maybe Integer
+baseStringToInt base = Just . getValue . reverse
+  where
+    getValue [] = 0
+    getValue (x:xs) = charValue x + base * getValue xs
+
+-- | Returns value of a single char
+charValue :: Char -> Integer
+charValue c
+    | num >= 48 && num <= 57 = num - 48
+    | num >= 97 && num <= 102 = num - 87
+    | num >= 65 && num <= 70 = num - 55
```

Comment:
I feel like `charValue` function does some redundant work. When using the `hexDigitChar` parser and similar, you already compare the char value with the number. I propose to implement custom parsers that do parsing and conversion. So you will be able to remove the `charValue` function completely. Something like this:

```haskell
hexDigitP :: Parser Integer
hexDigitP = hexCharToDigit <$> hexDigitChar

octalDigitP :: ... similar  goes here
```

And then conversion function will work with `[Integer]` instead of `String` (aka `[Char]`).